discovery/spec.md

```diff
@@ -55,305 +45,194 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+Discuss uri vs self properties....
 
-#### Intermediary [[CE](../cloudevents/spec.md#intermediary)]
-
-An "intermediary" receives a message containing an event for the purpose of
-forwarding it to the next receiver, which might be another intermediary or a
-[Consumer](#consumer-ce). A typical task for an intermediary is to route the
-event to receivers based on the information in the event
-[Context](../cloudevents/spec.md#context).
-
-#### Consumer [[CE](../cloudevents/spec.md#consumer)]
-
-A "consumer" receives the event and acts upon it. It uses the context and data
-to execute some logic, which might lead to the occurrence of new events.
-
-#### Subscription
-
-The request for events from a Service.
 
 ## Resource Model
 
-This section defines the resource model of a Discovery Endpoint.
-
-### Services
+This section defines the resource model of a Discovery Service.
 
-At the core of the data model is the concept of a [Service](#service). The API
-then exposes multiple ways to query over a list of Services. To help explain the
-Service resource, the following non-normative pseudo json shows its basic
-structure:
+### Endpoint
 
-(`*` means zero or more, `+` means one or more, `?` means optional)
-
-Service:
+An Endpoint is ...
 
+The following pseudo JSON shows the basic defintion of an Endpoint:
 ```
 {
-  "id": "[a unique URI segment within the authority]",
-  "authority": "[URI reference to the authority providing the service]", ?
-  "epoch": "[discovery entry epoch value]",
-  "name": "[unique name for this services]",
-  "url": "[unique URL to this service]",
-  "description": "[human string]", ?
-  "docsurl": "[URL reference for human documentation]", ?
-  "deprecated": { ?
-	"effectivetime": "[RFC3339 timestamp of when the service will enter a deprecated state]", ?
-    "removaltime": "[RFC3339 timestamp of when service will be removed]", ?
-    "alternative": "[URL to possible alternative service]", ?
-    "docsurl": "[URL reference to additional information]" ?
-  },
-  "specversions": [ "[ce-specversion value]" + ],
-  "subscriptionurl": "[URL to which the Subscribe request will be sent]",
-  "subscriptionconfig": { ?
-    "[key]": "[type]", *
-  },
-  "subscriptiondialects": [ "[dialect]" ], ?
-  "authscope": "[string]", ?
-  "protocols": [ "[string]" + ],
-  "events": [ ?
-    { *
-      "type": "[ce-type value]",
-      "description": "[human string]", ?
-      "datacontenttype": "[ce-datacontenttype value]", ?
-      "dataschema": "[ce-dataschema URI]", ?
-      "dataschematype": "[string per RFC 2046]", ?
-      "dataschemacontent": "[schema]", ?
-      "sourcetemplate": "[URI template per RFC 6570, level 1]", ?
-      "extensions": [ ?
-        { *
-          "name": "[CE context attribute name]",
-          "type": "[CE type string]",
-          "specurl": "[URL to specification defining the extension]" ?
-        }
-      ]
-    }
-  ]
+    "id": "URI-reference",
+    "name": "STRING",
+    "self": "URI",
```

Comment:
Yes I think the intent was to have `id` and `self` have the same endings. Would `self` always look the same for all clients or could the leading part of the self URI change based on how the client is accessing the system (e.g. public URL vs private URL)? I kind of assumed it might change at times, but `id` would always be static. WDYT?